x-pack/plugins/data_visualizer/public/application/index_data_visualizer/locator/locator.ts

```diff
@@ -94,6 +107,17 @@ export class IndexDataVisualizerLocatorDefinition
       appState.searchString = query.searchString;
       appState.searchQueryLanguage = query.searchQueryLanguage;
     }
+
+    if (params.searchString) {
+      appState.searchQuery = params.searchQuery;
+      appState.searchString = params.searchString;
+      appState.searchQueryLanguage = params.searchQueryLanguage;
+    }
+
+    if (filters) {
```

Comment:
The `samplerShardSize` is stored in the session, but on opening the session the control in the data visualizer page shows as the default `5000`. Could the setting be restored in the control?

Some of the other properties in the search session configuration are also not restored in the view - tested only `samplerShardSize` and `sortField`:

```
  "samplerShardSize": 5000,
  "pageIndex": 0,
  "indexPatternId": "9eeba8b0-5d94-11ec-aee9-43f51679c94a",
  "searchQuery": {
    "bool": {
      "filter": [],
      "must_not": [],
      "must": [
        {
          "match_all": {}
        }
      ]
    }
  },
  "sortField": "stats.cardinality",
  "visibleFieldNames": [],
  "timeRange": {
    "from": "2020-09-01T00:00:27.000Z",
    "to": "2021-08-31T23:53:40.000Z"
  },
  "showDistributions": true
```

Reply:
Tested https://github.com/elastic/kibana/pull/121053/commits/94fed45e2b8c8abc31f1c6f96bda964f7f0937f7 and confirmed `samplerShardSize` and `sortField` are restored correctly.